File: frontend/src/services/apiKeyService.js
import api from './api';
export const listApiKeys = () => api.get('/api/v1/api-keys');
export const createApiKey = (name) => api.post('/api/v1/api-keys', { name });
export const revokeApiKey = (id) => api.delete(`/api/v1/api-keys/${id}`);

File: backend/src/main/java/com/webhookplatform/service/ApiKeyService.java
package com.webhookplatform.service;

import com.webhookplatform.dto.request.CreateApiKeyRequest;
import com.webhookplatform.dto.response.ApiKeyResponse;

import java.util.List;
import java.util.UUID;

public interface ApiKeyService {
    ApiKeyResponse createApiKey(UUID userId, CreateApiKeyRequest request);
    List<ApiKeyResponse> listApiKeys(UUID userId);
    void revokeApiKey(UUID keyId, UUID userId);
}
